c10/mobile/CPUProfilingAllocator.cpp

```diff
@@ -205,7 +216,11 @@ std::vector<uint64_t> formulate_greedy_allocation_plan(
         auto merge_block_iter = end_it->second;
         auto merge_block_size = merge_block_iter->first;
         freed_size += merge_block_size;
-        free_size_to_offset.erase(merge_block_iter);
+        if (merge_block_iter->second.size() == 1) {
+          free_size_to_offset.erase(merge_block_iter);      
+        } else {
+          merge_block_iter->second.erase(std::remove(merge_block_iter->second.begin(), merge_block_iter->second.end(), end_it->first), merge_block_iter->second.end());
```

Comment:
Should this be removing only the matching offset, that is end_it->first? Which is what you are removing via `std::remove`. Then what is teh `merge_block_iter->second.erase` for?

Reply:
https://stackoverflow.com/questions/39912/how-do-i-remove-an-item-from-a-stl-vector-with-a-certain-value